v3/conn.go

```diff
@@ -145,35 +148,87 @@ type DialContext struct {
 	tc *tls.Config
 }
 
-func (dc *DialContext) dial(u *url.URL) (net.Conn, error) {
+func (dc *DialContext) dial(u *url.URL) (conn net.Conn, err error) {
 	if u.Scheme == "ldapi" {
 		if u.Path == "" || u.Path == "/" {
 			u.Path = "/var/run/slapd/ldapi"
 		}
 		return dc.d.Dial("unix", u.Path)
 	}
 
-	host, port, err := net.SplitHostPort(u.Host)
-	if err != nil {
-		// we assume that error is due to missing port
-		host = u.Host
-		port = ""
+	if u.Scheme != "ldap" && u.Scheme != "ldaps" {
+		return nil, fmt.Errorf("Unknown scheme '%s'", u.Scheme)
 	}
 
-	switch u.Scheme {
-	case "ldap":
-		if port == "" {
-			port = DefaultLdapPort
+	hostports := make([]string, 0, 1)
+
+	if u.Host == "" {
+		fragments := strings.Split(u.Path[1:], ",")
+		pieces := make([]string, 0, len(fragments))
+		for _, fragment := range fragments {
+			if strings.HasPrefix(fragment, "dc=") {
+				pieces = append(pieces, fragment[3:])
+			}
 		}
-		return dc.d.Dial("tcp", net.JoinHostPort(host, port))
-	case "ldaps":
+
+		domain := strings.Join(pieces, ".")
+		_, records, err := net.LookupSRV("ldap", "tcp", domain)
+		if err != nil {
+			return nil, err
+		}
+
+		sort.Slice(records, func(i, j int) bool {
+			return records[i].Priority > records[j].Priority
+		})
+
+		if u.Scheme == "ldaps" {
+			dc.tc = &tls.Config{
+				ServerName: domain,
+			}
+		}
+
+		for _, record := range records {
+			port := strconv.Itoa(int(record.Port))
+			hostports = append(hostports, net.JoinHostPort(record.Target, port))
+		}
+	} else {
+		host, port, err := net.SplitHostPort(u.Host)
+		if err != nil {
+			// we assume that error is due to missing port
```

Comment:
May you check for the error explictly? [An IPv6 address for example can still be borged](https://cs.opensource.google/go/go/+/refs/tags/go1.13:src/net/ipsock.go;l=156)

Reply:
It was from the original code, if it needs to be fixed, perhaps it could be in another PR?